### PR TITLE
feat: add examples implementation and documentation

### PR DESCRIPTION
**Pull request by rubenVB01**
## Description

I added 3 examples with their respective documentation:

- An example where the module creates a bucket but the user does not provide the name
- An example where the module creates a bucket with the name provided by the user
- An example where the module uses an existing bucket

### CODE REVIEW

File: examples/s3bucket-create_bucket_generate_name/variables.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  description = "AWS private key."
+  type        = string
+}
+
+variable "aws_access_key" {
+  description = "AWS access key."
+  type        = string
+}
+
+variable "region" {
```
Comment (EspenAlbert) [inline examples/s3bucket-create_bucket_generate_name/variables.tf]: I would use `aws_region` to be consistent with other variables

File: examples/s3bucket-create_bucket_generate_name/variables.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  description = "Atlas project id."
+  type        = string
+}
+
+variable "aws_secret_access_key" {
+  description = "AWS private key."
+  type        = string
+}
+
+variable "aws_access_key" {
```
Comment (EspenAlbert) [inline examples/s3bucket-create_bucket_generate_name/variables.tf]: I would advice against using explicit variables for `aws_access_key` and `aws_secret_access_key`. It is more convenient to configure these with env-vars

File: examples/s3bucket-create_bucket_provided_name/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# push-based-log-export-s3_bucket - create a new bucket with a provided name
+
+This example shows how you can use the push-based-log-export-s3_bucket submodule to enable push-based logging in an Atlas project using a new s3 bucket that has a name provided by yourself.
+
+
+## Usage
+
+1. Set the following variables: 
+
```
Comment (EspenAlbert) [inline examples/s3bucket-create_bucket_provided_name/README.md]: Add instructions on how to update the name in `main.tf`?

File: examples/s3bucket-use_provided_bucket/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# push-based-log-export-s3_bucket - use an existing bucket
+
+This example shows how you can use the push-based-log-export-s3_bucket submodule to enable push-based logging in an Atlas project using an existing s3 bucket.
+
+
+## Usage
```
Comment (EspenAlbert) [inline examples/s3bucket-use_provided_bucket/README.md]: Add instructions to update `main.tf`?

File: examples/s3bucket-create_bucket_generate_name/README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# push-based-log-export-s3_bucket - create a new bucket with a default name
```
Comment (AgustinBettati) [inline examples/s3bucket-create_bucket_generate_name/README.md]: Both for the title and name of directories in examples would be consistent with using snake_case or dash-case. Personally like more dash case and see it used in [aws module](https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/tree/master/examples) as well.

File: examples/s3bucket-create_bucket_generate_name/README.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+This example shows how you can use the push-based-log-export-s3_bucket submodule to enable push-based logging in an Atlas project using a new s3 bucket that has a default name.
```
Comment (AgustinBettati) [inline examples/s3bucket-create_bucket_generate_name/README.md]:
```suggestion
This example shows how you can use the s3 bucket submodule to enable push-based logging in an Atlas project. The module will create a new s3 bucket with a default name.
```
Just making more explicit the fact that the s3 bucket is being created as part of the module. Suggestion also applies to second example.

File: examples/s3bucket-create_bucket_generate_name/variables.tf (hunk 2 above)
Comment (rubenVB01) [reply examples/s3bucket-create_bucket_generate_name/variables.tf]: I was trying to put the same way as in the mongodbatlas provider, but if it's better to configure these with env vars I can change it and give instructions to set the env variables

**Comment by rubenVB01** [comment]
Applied all the suggestions in the PR comments. Also added the README for the tests (forgot to add it in the previous PR)

File: examples/s3bucket-create_bucket_generate_name/variables.tf (hunk 2 above)
Comment (AgustinBettati) [reply examples/s3bucket-create_bucket_generate_name/variables.tf]: dont have a strong opinion but would keep consistency with how atlas keys are also provided. push based logging example provides both as env variables ([reference](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/master/examples/mongodbatlas_push_based_log_export#usage))

File: examples/s3bucket-create-bucket-provided-name/README.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
+# push-based-log-export-s3-bucket - create a new bucket with a provided name
+
+This example shows how you can use the s3 bucket submodule to enable push-based logging in an Atlas project. The module will create a new s3 bucket with a default name.
```
Comment (AgustinBettati) [inline examples/s3bucket-create-bucket-provided-name/README.md]:
```suggestion
This example shows how you can use the s3 bucket submodule to enable push-based logging in an Atlas project. The module will create a new s3 bucket with the name specified.
```

**Review by AgustinBettati — APPROVED** [review]
LGTM, thanks for follow ups

**Pull request merged**